app/models/entry.rb

```diff
@@ -77,4 +77,8 @@ def previous_entry
   def next_entry
     Entry.where(course: course, level: 3).where(Entry.arel_table[:sequence].gt(sequence)).first
   end
+
+  def self.search(q)
```

Comment:
This is very good! 💯 
Please place this class method before instance methods (this is not required in general, but it is only style guide used in this project.